core/src/main/java/com/segment/analytics/kotlin/core/Settings.kt

```diff
@@ -79,6 +104,7 @@ suspend fun Analytics.checkSettings() {
                 log("Dispatching update settings on ${Thread.currentThread().name}")
                 store.dispatch(System.UpdateSettingsAction(settingsObj), System::class)
                 update(settingsObj, updateType)
+                store.dispatch(System.ToggleSettingsDispatch(dispatched = true), System::class)
```

Comment:
just to confirm: we dispatch twice here because we want to ensure settings being successfully propagated to subscribers and then toggle the state?

Reply:
yup i didnt want to preemptively toggle the dispatch